### ISSUE TITLE
BLOCKED ON ISSUE 139 ## Are there any examples of the waveform progress updating as an audio file plays?

### ISSUE THREAD

**Issue by sumitpaul** [fulldecent/FDWaveformView#130]
I looked at the "Examples" folder in the repo but it's not very clear how this would work with a playing audio. I am able to display the waveform and I can tap to change the highlighted part of the waveform, but I can't figure out how to connect this progress to the playing audio file. Please help.

**Comment by ibilalnawaz** [comment]
Any progress ?

**Comment by rthompson40** [comment]
Hi there,

I know this is probably no longer needed by either of you but perhaps this can help a future user:

1. In viewDidLoad(), add your FDWaveformView as per your specification (in this example, I used interface builder and my IBOutlet is called waveView). Ensure scrolling is enabled for your waveform:

![image](https://user-images.githubusercontent.com/65298685/86953631-96ce5000-c14c-11ea-86a8-d791f3e82593.png)

2. I'm using a button to trigger playback in this example.  Upon pressing the button when no file is playing, I call this function. The animation time parameter is set to equal the length of the audio file for fairly obvious reasons. (NOTE: without the .curveLinear option, the 'scroll' will be out-of-sync. This was unfortunately not apparent in the example project but thankfully I realised it was a non-linear response after some trial and error): 

![image](https://user-images.githubusercontent.com/65298685/86953794-d5640a80-c14c-11ea-8106-da9fc87c9cff.png)

3. Upon finishing the audio playback, I reset the 'progress' by triggering this method:

![image](https://user-images.githubusercontent.com/65298685/86954357-be71e800-c14d-11ea-8d5a-71c10133db94.png)

Hopefully this helps! I'm fairly new to software development so forgive me if there's any glaring 'no-no's.

**Comment by fulldecent** [comment]
You got it I've added this to the example app and you can see it here

https://github.com/user-attachments/assets/9b74f788-9fc3-4d5b-8d71-f67e821819d1
